.github/workflows/amis-workflow.yml

```diff
@@ -0,0 +1,157 @@
+---
+name: amis-workflow
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - 'v*'
+  schedule:
+    - cron: "0 0 * * 0"
+  # Add pull_request for testing
+  pull_request:
+    branches:
+      - master
+jobs:
+  publish-amis-to-marketplace:
+    # needs:
+    #  - release-hold
+    #  - ami-hold
+    if: ${{ needs.path_filter.outputs.should_skip != 'true' }}
+    name: publish-amis-to-marketplace job
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+      CODE_DIR: "${{ github.workspace }}/experimental/cloudstrapper"
+      VARS_DIR: "${{ github.workspace }}/experimental/cloudstrapper/playbooks/roles/vars"
+      WORK_DIR: "${{ github.workspace }}/experimental/cloudstrapper/playbooks"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run apt
+        run: sudo apt-get update && sudo apt -y upgrade
+      - name: setup pyenv
+        uses: "gabrielfalcao/pyenv-action@v8"
+        with:
+          default: 3.7.0
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - name: Install Dependencies
+        run: |
+            pip install ansible awscli boto3
+            sudo apt-get update
+      - name: Determine AWS credentials to use
+        run: |
+            if ${{ github.ref == 'refs/heads/v1.*' && github.event_name != 'schedule' }}; then
+              echo AWS_ACCOUNT="LF" >> $GITHUB_ENV
+            elif ${{ github.event_name == 'schedule' }}; then
+              echo AWS_ACCOUNT="FB" >> $GITHUB_ENV
+            fi
+            # Default to FB for testing, delete this line before merging
+            echo AWS_ACCOUNT="FB" >> $GITHUB_ENV
+      - name: Propagate AWS credentials to ansible and create version
+        run: |
+            if [ "${{ env.AWS_ACCOUNT }}" = "FB" ]; then
+              sed -i -e "s@awsAccessKey:@& ${{ secrets.FB_AWS_ACCESS_KEY }}@1"  ${{ env.VARS_DIR }}/secrets.yaml
+              sed -i -e "s@awsSecretKey:@& ${{ secrets.FB_AWS_SECRET_ACCESS_KEY }}@1"  ${{ env.VARS_DIR }}/secrets.yaml
+              echo VERSION="1.6.0-${{ github.sha }}" >> $GITHUB_ENV
+              echo PACKAGE_VERSION="1.6.0" >> $GITHUB_ENV
+              # When focal-ci is clean we will be able to dynamically fetch latest ci debian packages
+            elif  [ "${{ env.AWS_ACCOUNT }}" = "LF" ]; then
+              sed -i -e "s@awsAccessKey:@& ${{ secrets.LF_AWS_ACCESS_KEY }}@1"  ${{ env.VARS_DIR }}/secrets.yaml
+              sed -i -e "s@awsSecretKey:@& ${{ secrets.LF_AWS_SECRET_ACCESS_KEY }}@1"  ${{ env.VARS_DIR }}/secrets.yaml
+              echo PACKAGE_VERSION="1.7.0" >> $GITHUB_ENV
```

Comment:
when triggered with workflow dispatch on a branch, could you parse the branch name v1.x.x to get 1.x.x here

Reply:
I wrote a potential way to do this, but will need to test